plugins/aggregators/derivative/derivative.go

```diff
@@ -0,0 +1,211 @@
+package derivative
+
+import (
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+)
+
+type Derivative struct {
+	Variable    string          `toml:"variable"`
+	Infix       string          `toml:"infix"`
+	MaxRollOver uint            `toml:"max_roll_over"`
+	Log         telegraf.Logger `toml:"-"`
+	cache       map[uint64]aggregate
+}
+
+type aggregate struct {
+	first    *event
+	last     *event
+	name     string
+	tags     map[string]string
+	rollOver uint
+}
+
+type event struct {
+	fields map[string]float64
+	time   time.Time
+}
+
+func NewDerivative() *Derivative {
+	derivative := &Derivative{Infix: "_by_", MaxRollOver: 10}
+	derivative.cache = make(map[uint64]aggregate)
+	derivative.Reset()
+	return derivative
+}
+
+var sampleConfig = `
+	## This Aggregator will estimate a derivative for each field, which is
+	## contained in both the first and last metric of the aggregation interval.
+	## Without further configuration the derivative will be calculated with
+	## respect to the time difference between these two measurements in seconds.
+	## The formula applied is for every field:
+	##
+	##               value_last - value_first
+	## derivative = --------------------------
+	##              time_difference_in_seconds
+	##
+	## The resulting derivative will be named *fieldname_by_seconds*. The infix
+	## "_by_" can be configured by the *infix* parameter.
+	# infix = "_by_"
+	##
+	## As an abstraction the derivative can be calculated not only by the time
+	## difference but by the difference of a field, which is contained in the
+	## measurement. This field is assumed to be monotonously increasing. This
+	## feature is used by specifying a *variable*.
+	# variable = "parameter"
```

Comment:
Can you please sync this with the README and use
```suggestion
	# variable = ""
```